eZ/Publish/Core/Repository/Helper/RelationProcessor.php

```diff
@@ -48,21 +51,42 @@ public function __construct(Handler $handler)
      * @param array $locationIdToContentIdMapping An array with Location Ids as keys and corresponding Content Id as values
      * @param \eZ\Publish\SPI\FieldType\FieldType $fieldType
      * @param \eZ\Publish\Core\FieldType\Value $fieldValue Accepted field value.
-     * @param string $fieldDefinitionId
+     * @param \eZ\Publish\API\Repository\Values\ContentType\FieldDefinition $fieldDefinition
+     *
+     * @return \eZ\Publish\SPI\FieldType\ValidationError[]
      */
     public function appendFieldRelations(
         array &$relations,
         array &$locationIdToContentIdMapping,
         SPIFieldType $fieldType,
         BaseValue $fieldValue,
-        $fieldDefinitionId
+        FieldDefinition $fieldDefinition
```

Comment:
1st you are commenting on a take that will likely not happen, keeping invalid relation with no further info to go with it will break content view _(preview)_. Secondly it's an internal Repo class so if it was needed I don't see the problem but I digress.